contracts/asset-proxy/contracts/src/MultiAssetProxy.sol

```diff
@@ -204,15 +228,20 @@ contract MultiAssetProxy is
                     let nestedAssetDataElementOffset := calldataload(add(nestedAssetDataContentsStart, i))
 
                     // In order to find the start of the `nestedAssetData[i]` contents, we must add:
-                    // 4 (function selector) 
-                    // + assetDataOffset 
+                    // assetDataOffset 
                     // + 32 (assetData len)
                     // + 4 (assetProxyId)
                     // + nestedAssetDataOffset
                     // + 32 (nestedAssetData len)
                     // + nestedAssetDataElementOffset
                     // + 32 (nestedAssetDataElement len)
-                    let nestedAssetDataElementContentsStart := add(assetDataOffset, add(nestedAssetDataOffset, add(nestedAssetDataElementOffset, 104)))
+                    let nestedAssetDataElementContentsStart := add(
+                        assetDataOffset,
+                        add(
+                            nestedAssetDataOffset,
+                            add(nestedAssetDataElementOffset, 100)
+                        )
+                    )
```

Comment:
It seems like there is a gas efficiency reason for not asserting that `nestedAssetDataContentsStart + nestedAssetDataLen <= assetDataEnd`, but I wonder if it would be worth it to include it just to ensure that the nestedAssetData is in 1:1 correspondence with the data that is actually sent.

Reply:
In general I'm all for extra validation, but I primarily want to get this exact version merged since it is the one deployed to mainnet.